drivers/volume/portworx/schedops/k8s-schedops.go

```diff
@@ -451,6 +451,22 @@ func (k *k8sSchedOps) UpgradePortworx(ociImage, ociTag string) error {
 	return nil
 }
 
+//Method to validate if Portworx pod is up and running
+func (k *k8sSchedOps) IsPXAppRunningOnNode(n node.Node) bool {
+	pxPods, err := k8s.Instance().GetPodsByNode(n.Name, PXNamespace)
+	if err != nil {
+		logrus.Errorf("Failed to get apps on node %s", n.Name)
+		return false
+	}
+	for _, pod := range pxPods.Items {
+		if strings.HasPrefix(pod.Name, "portworx") && pod.Status.Phase != "Running" {
```

Comment:
Instead of checking pod.name check pod labels. PX pods have label `name=portworx`.

Reply:
Fixed.